src/App.js

```diff
@@ -1,51 +1,119 @@
 import styled from 'styled-components/macro'
+import {Switch, Route, useHistory} from 'react-router-dom'
 import useGameData from './hooks/useGameData'
-import SaveGameForm from './components/SaveGameForm'
-import GameCardsList from './components/GameCardsList'
+import useForm from './hooks/useForm'
+import Header from './components/layout/Header'
+import Footer from './components/layout/Footer'
+import SaveGamePage from './SaveGamePage'
+import GameCardsPage from './GameCardsPage'
 
 function App() {
 
   const { 
-    targetProfile, 
-    savedGameProfiles, 
-    isEditFormShown, 
-    addGameProfile, 
-    deleteGameProfile, 
-    editGameProfile, 
-    prepareEditModus, 
-    cancelEditModus } = useGameData()
+      targetProfile, 
+      savedGameProfiles, 
+      isEditFormShown, 
+      addGameProfile, 
+      deleteGameProfile, 
+      editGameProfile, 
+      prepareEditModus, 
+      cancelEditModus } = useGameData()
   
+    const {
+        formInputs, 
+        showSaveButton,
+        updateDirtyInputs,
+        handleChange,
+        showErrorMessage,
+        handleSubmit,
+        handleCancelEditModus,
+        resetForm } = useForm({
+                              targetProfile,
+                              isEditFormShown,
+                              addGameProfile,
+                              editGameProfile,
+                              cancelEditModus,
+                              showGameCardsPage})
+
+  const history = useHistory()
 
   return (
     <AppWrapper>
-      {isEditFormShown ? 
-        <SaveGameForm 
-            onSubmit={addGameProfile} 
-            isEditFormShown={isEditFormShown} 
-            targetProfile={targetProfile} 
-            editGameProfile={editGameProfile}
-            cancelEditModus={cancelEditModus}/>
-      : 
-      <SaveGameFormWrapper>
-        <SaveGameForm onSubmit={addGameProfile} />
-        <GameCardsList 
-            savedGameProfiles={savedGameProfiles} 
-            onDelete={deleteGameProfile} 
-            onEdit={prepareEditModus}/>
-      </SaveGameFormWrapper>
-      }
+      <HeaderWrapper>
```

Comment:
I used the ``HeaderWrapper``, because otherwise the components would be accordingly to their styling placed at the top / bottom in styleguidist. This way it also works in styleguidist. :)